### PR TITLE
provider/aws: Add JSON validation to the aws_kms_key resource.

### PR DESCRIPTION
**Pull request by kwilczynski**
This commit adds support for new helper function which is used to
normalise and validate JSON string.

Signed-off-by: Krzysztof Wilczynski <krzysztof.wilczynski@linux.com>

### CODE REVIEW

**Comment by kwilczynski** [comment]
Part of work which began in #8028.

**Comment by kwilczynski** [comment]
Acceptance test is passing:

```
$ make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSKmsKey_'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/09/14 17:30:35 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSKmsKey_ -timeout 120m
=== RUN   TestAccAWSKmsKey_importBasic
--- PASS: TestAccAWSKmsKey_importBasic (46.64s)
=== RUN   TestAccAWSKmsKey_basic
--- PASS: TestAccAWSKmsKey_basic (56.70s)
=== RUN   TestAccAWSKmsKey_policy
--- PASS: TestAccAWSKmsKey_policy (46.68s)
=== RUN   TestAccAWSKmsKey_isEnabled
--- PASS: TestAccAWSKmsKey_isEnabled (527.79s)
PASS
ok     	github.com/hashicorp/terraform/builtin/providers/aws   	677.830s
```

File: builtin/providers/aws/resource_aws_kms_key.go [hunk 1]
```diff
@@ -143,7 +144,8 @@ func resourceAwsKmsKeyRead(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	d.Set("policy", normalizeJson(*p.Policy))
+	policy, _ := normalizeJsonString(*p.Policy)
```
Comment (stack72) [inline builtin/providers/aws/resource_aws_kms_key.go]: Do we need to check for any errors here?

**Comment by stack72** [comment]
LGTM! Thanks for the work here :)

```
% make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSKmsKey_'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/09/22 09:16:33 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSKmsKey_ -timeout 120m
=== RUN   TestAccAWSKmsKey_importBasic
--- PASS: TestAccAWSKmsKey_importBasic (56.41s)
=== RUN   TestAccAWSKmsKey_basic
--- PASS: TestAccAWSKmsKey_basic (68.63s)
=== RUN   TestAccAWSKmsKey_policy
--- PASS: TestAccAWSKmsKey_policy (55.11s)
=== RUN   TestAccAWSKmsKey_isEnabled
--- PASS: TestAccAWSKmsKey_isEnabled (298.81s)
PASS
ok     	github.com/hashicorp/terraform/builtin/providers/aws   	478.973s
```

**Pull request merged**